lua/entities/gmod_wire_expression2/core/e2lib.lua

```diff
@@ -296,6 +296,19 @@ end
 
 -- ----------------------------- compiler stuf ---------------------------------
 
+-- A function suitable for use as xpcall's error handler. If the error is
+-- generated by Compiler:Error, Parser:Error, etc., then the string will be a
+-- usable error message. If not, then it's an error not caused by an error in
+-- user code, and so we dump a stack trace to the console to help debug it.
+function E2Lib.errorHandler(message)
+	if string.match(message, " at line ") then return message end
```

Comment:
wouldnt that be the "caused by" thing?
couldnt you do common substring elimination like java does and add the "caused by" after the previous error?